openverse-api/catalog/api/serializers/audio_serializers.py

```diff
@@ -0,0 +1,177 @@
+    MediaSerializer,
+    AboutMediaSerializer,
+)
+
+
+class AudioSearchQueryStringSerializer(MediaSearchQueryStringSerializer):
+    """ Parse and validate search query string parameters. """
+
+    fields_names = [
+        *MediaSearchQueryStringSerializer.fields_names,
+        'source',
+        'categories',
+        'duration',
+    ]
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+
+    source = serializers.CharField(
+        label="provider",
+        help_text="A comma separated list of data sources to search. Valid "
+                  "inputs: "
+                  f"`{list(get_sources('audio').keys())}`",
+        required=False
+    )
+    categories = serializers.CharField(
+        label="categories",
+        help_text="A comma separated list of categories; available categories "
+                  "include `music`, `sound_effect`, `podcast`, `audiobook`, "
+                  "and `news`.",
+        required=False
+    )
+    duration = serializers.CharField(
+        label='duration',
+        help_text="A comma separated list of audio lengths; available lengths "
+                  "include `short`, and `long`.",
+        required=False
+    )
+
+    @staticmethod
+    def validate_source(input_sources):
+        allowed_sources = list(get_sources('audio').keys())
+        input_sources = input_sources.split(',')
+        input_sources = [x for x in input_sources if x in allowed_sources]
+        input_sources = ','.join(input_sources)
+        return input_sources.lower()
+
+    @staticmethod
+    def validate_categories(value):
+        valid_categories = {
+            'music',
+            'sound_effect',
+            'podcast',
+            'news',
+            'audiobook',
+        }
+        _validate_enum('category', valid_categories, value)
+        return value.lower()
+
+    @staticmethod
+    def validate_duration(value):
+        valid_ratios = {  # TODO: Finalise duration filters
+            'short',
+            'long'
+        }
+        _validate_enum('aspect ratio', valid_ratios, value)
+        return value.lower()
+
+
+class AudioSerializer(MediaSerializer):
+    """ A single audio file. Used in search results."""
+
+    fields_names = [
+        *MediaSerializer.fields_names,
+        'audio_set',
+        'genre',
+        'duration',
+        'bit_rate',
+        'sample_rate',
+        'alt_files',
+        'detail_url',
+        'related_url',
+    ]
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+
+    audio_set = serializers.PrimaryKeyRelatedField(
+        required=False,
+        help_text='Reference to set of which this track is a part.',
+        read_only=True
+    )
+
+    genres = serializers.ListField(
+        child=serializers.CharField(),
+        required=False,
+        help_text=''
```

Comment:
```suggestion
        help_text='An array of audio genres such as `rock`, `electronic` for `music` category, or `politics`, `sport`, `education` for `podcast` category'
```

Reply:
Resolved in ddba0a4. Couldn't commit suggestion directly because of line length.